api/proto/teleport/workloadidentity/v1/revocation_service.proto

```diff
@@ -0,0 +1,112 @@
+  // revocation by name.
+  rpc GetWorkloadIdentityX509Revocation(GetWorkloadIdentityX509RevocationRequest) returns (WorkloadIdentityX509Revocation);
+  // DeleteWorkloadIdentityX509Revocation deletes a X509 workload identity
+  // revocation by name.
+  rpc DeleteWorkloadIdentityX509Revocation(DeleteWorkloadIdentityX509RevocationRequest) returns (google.protobuf.Empty);
+  // ListWorkloadIdentityX509Revocations returns a list of X509 workload
+  // identity revocations, pagination semantics are applied.
+  rpc ListWorkloadIdentityX509Revocations(ListWorkloadIdentityX509RevocationsRequest) returns (ListWorkloadIdentityX509RevocationsResponse);
+  // StreamSignedCRL returns the latest signed CRL to the client, and, any
+  // later updates to this signed CRL.
+  rpc StreamSignedCRL(StreamSignedCRLRequest) returns (stream StreamSignedCRLResponse);
+}
+
+// The request for CreateWorkloadIdentityX509Revocation.
+message CreateWorkloadIdentityX509RevocationRequest {
+  // The X509 workload identity revocation to create.
+  WorkloadIdentityX509Revocation workload_identity_x509_revocation = 1;
+}
+
+// The request for UpdateWorkloadIdentityX509Revocation.
+message UpdateWorkloadIdentityX509RevocationRequest {
+  // The X509 workload identity revocation to update.
+  WorkloadIdentityX509Revocation workload_identity_x509_revocation = 1;
+}
+
+// The request for UpsertWorkloadIdentityX509Revocation.
+message UpsertWorkloadIdentityX509RevocationRequest {
+  // The X509 workload identity revocation to upsert.
+  WorkloadIdentityX509Revocation workload_identity_x509_revocation = 1;
+}
+
+// The request for GetWorkloadIdentityX509Revocation.
+message GetWorkloadIdentityX509RevocationRequest {
+  // The name of the X509 workload identity revocation to retrieve.
+  string name = 1;
+}
+
+// The request for DeleteWorkloadIdentityX509Revocation.
+message DeleteWorkloadIdentityX509RevocationRequest {
+  // The name of the workload identity to delete.
+  string name = 1;
+}
+
+// The request for ListWorkloadIdentityX509Revocations.
+message ListWorkloadIdentityX509RevocationsRequest {
+  // The maximum number of items to return.
+  // The server may impose a different page size at its discretion.
+  int32 page_size = 1;
+  // The page_token value returned from a previous ListWorkloadIdentities request, if any.
+  string page_token = 2;
+}
+
+// The response for ListWorkloadIdentityX509Revocations.
+message ListWorkloadIdentityX509RevocationsResponse {
+  // The page of workload identities that matched the request.
+  repeated WorkloadIdentityX509Revocation workload_identity_x509_revocations = 1;
+  // Token to retrieve the next page of results, or empty if there are no
+  // more results in the list.
+  string next_page_token = 2;
+}
+
+// The request for StreamSignedCRL.
+message StreamSignedCRLRequest {}
+
+// The response for StreamSignedCRL.
+message StreamSignedCRLResponse {
+  // The signed Certificate Revocation List (CRL). Encoded in DER format.
```

Comment:
Can you better specify the CRL format, for the non-initiated?

Is the signature appended/prepended in the bytes itself, instead of being a separate field? Is that part of the CRL format spec too?

Reply:
f3b114b92e